src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -588,20 +589,41 @@ def __deepcopy__(self, memodict={}) -> "SelectableEntity":  # noqa: B006
 
     @property
     def sql_query(self) -> str:
-        return f"{analyzer_utils.SELECT}{analyzer_utils.STAR}{analyzer_utils.FROM}{self.entity.name}"
+        table_reference = self.entity.name
+
+        if self.entity.time_travel_config is not None:
+            table_reference += generate_time_travel_sql_clause(
+                self.entity.time_travel_config
+            )
```

Comment:
can this be calculated in the constructor so that we avoid recompute each time we call the property?